### DISCUSSION TITLE
Website demos are using Fabricjs 5.3.x and I spent hours trying to figure out why image upload of PNG and SVG was not working

### DISCUSSION THREAD

**Discussion by designlook** [fabricjs/fabric.js#10134] [General]
I was looking at the code here in this demo
http://fabricjs.com/clipping

And re-used code to build an uploader and spent hours and could not figure out why it was not working. There was no errors in console and I spent hours trying different things until I stumbled upon [this discussion](https://github.com/fabricjs/fabric.js/issues/10020) today and finally understood why the code was not working (the demo was actually using fabricjs 5.x code). There was a link to a stackblitz in that discussion and the code there finally clued me in on why my code was not working. It needs the then(img)

```
fabric.FabricImage.fromURL(iconUrl)
      .then((img) => {
        img.set({
          left: pointer.x,
          top: pointer.y,
          width: 50, // Adjust dimensions as needed
          height: 50,
          selectable: true, // Make icon selectable
          evented: false, // Disable events (like drag)
          hasControls: false, // Disable controls (like rotating)
        });

        canvas.add(img); // Add image to canvas
        img.setCoords(); // Set coordinates for precise positioning
        canvas.bringObjectToFront(img); // Bring the icon to the front
        canvas.renderAll(); // Render canvas after adding the image

        // Check if the image is added to the canvas
        const objectsOnCanvas = canvas.getObjects();
        const isImageAdded = objectsOnCanvas.includes(img);

        if (isImageAdded) {
          console.log('Image successfully added to canvas:', img);
        } else {
          console.error('Failed to add image to canvas.');
        }
      })
      .catch((err) => console.error('Error loading icon:', err));
  }
```

It would be really helpful if demos included a demo with fabricjs 6.x where it shows an upload button as well as input to a URL and show how upload of png, jpeg, gif would work via upload as well as via a URL.

Also would be great if the demos could be labeled with what version of fabricjs it's running. I was assuming all the demos were all using version 6.x

**Comment by asturur** [comment]
Migrating the website is taking long, there are updated demos here:
https://fabricjs.github.io

**Reply by asturur** [reply]
the img url is still broken, but if you change it to
const demoImg = 'https://fabricjs.github.io//assets/pug.jpg'; it will work

**Reply by designlook** [reply]
Thanks so much! Much appreciate your fantastic work on fabricjs!!
